### PR TITLE
File emulator quantization

### PR DESCRIPTION
**Pull request by jlashner**
Makes several changes to the Smurf File Emulator so that it's better for testing the bookbinder.

## Description

- Adds "timestamp quantization".  By clamping the start / stop times to integers, `np.arange` with an integer sample freq will create close to co-sampled data, even if the start / stop times are different between streams.
- Adds correct tags to the stream function
- Adds the ability to set a drop-chance, which will toss a subset of data points.
- Adds some params that tell the `stream` task to use the `stream_between` function of the streamer, instead of generating frames one-by-one, allowing for the task to run for a non-zero duration without sleeping.

## Motivation and Context

This is required to properly test the bookbinder on the emulated data.

## How Has This Been Tested?

Tested by creating sample data and running it all the way through to books.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: socs/agents/smurf_file_emulator/agent.py [hunk 1]
```diff
@@ -475,7 +505,8 @@ def uxm_setup(self, session, params):
         """
         self._run_setup(action='uxm_setup', action_time=time.time(),
                         session=session)
-        time.sleep(1)
+        if params.get('sleep', True):
```
Comment (BrianJKoopman) [inline socs/agents/smurf_file_emulator/agent.py:508]: Create a `Parameters:` section in the docstring with the new 'sleep' parameter and the old 'test_mode' parameter.

(`@ocs_agent.param` decorators could also be used.)

**Review by BrianJKoopman — CHANGES_REQUESTED** [review]
This looks good! Just one docstring related request.

File: socs/agents/smurf_file_emulator/agent.py (hunk 1 above)
Comment (jlashner) [reply socs/agents/smurf_file_emulator/agent.py:508]: Not sure where test_mode was copied from, but it's not actually implemented here... so I just removed from the spec line.

**Comment by jlashner** [comment]
Great, thanks brian! Fixed the uxm_setup docstring.

**Review by BrianJKoopman — APPROVED** [review]
Awesome, looks good!

**Pull request merged**